### PR TITLE
Update search response with optionalibility

### PR DESCRIPTION
**Pull request by Mpdreamz**
Reduces failures down to 15 locally.

The remainder are aggregations N level deep for which we need to
formulate a battle plan.

### CODE REVIEW

File: specification/specs/search/search/hits/Hit.ts [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-  _version: long;
-  sort: Array<number | string>;
+
+  _score?: double;
+  _type?: TypeName;
+
+  _explanation?: Explanation;
+  fields?: Dictionary<string, LazyDocument>;
+  highlight?: Dictionary<string, string[]>;
+  /** @prop_serializer VerbatimInterfaceReadOnlyDictionaryKeysFormatter`2 */
```
Comment (delvedor) [inline specification/specs/search/search/hits/Hit.ts:11]: Should we keep this?

File: specification/specs/search/search/hits/Hit.ts [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
+  _ignored?: string[];
+
+  _shard?: string;
+  _node?: string;
+  _routing?: string;
+  _source?: TDocument;
+  _seq_no?: long;
+  _primary_term?: long;
+  _version?: long;
+  sort?: Array<number | string>;
```
Comment (delvedor) [inline specification/specs/search/search/hits/Hit.ts:24]: We should try to be as much specific as possible, is this a `long` or a `double`?

File: specification/specs/aggregations/Aggregate.ts [hunk 3]
```diff
@@ -67,7 +72,7 @@ class AutoDateHistogramAggregate extends AggregateBase {
 }
 
 class FiltersAggregate extends AggregateBase {
-  buckets:FiltersBucketItem[];
+  buckets:FiltersBucketItem[] | Dictionary<string, FiltersBucketItem>;
```
Comment (delvedor) [inline specification/specs/aggregations/Aggregate.ts:75]:
```suggestion
  buckets: FiltersBucketItem[] | Dictionary<string, FiltersBucketItem>;
```

File: output/typescript/types.ts [hunk 4]
```diff
@@ -6135,35 +6138,37 @@ export interface HighlightField {
...
+  _nested?: NestedIdentity
+  _ignored?: Array<string>
+  _shard?: string
+  _node?: string
+  _routing?: string
+  _source?: TDocument
+  _seq_no?: long
+  _primary_term?: long
+  _version?: long
+  sort?: Array<number | string>
```
Comment (delvedor) [inline output/typescript/types.ts:6159]: Same here.

File: output/typescript/types.ts [hunk 5]
```diff
@@ -139,7 +143,7 @@ export interface ExtendedStatsAggregate extends StatsAggregate {
 }
 
 export interface FiltersAggregate extends AggregateBase {
-  buckets: Array<FiltersBucketItem>
+  buckets: Array<FiltersBucketItem> | Record<string, FiltersBucketItem>
```
Comment (delvedor) [inline output/typescript/types.ts:146]: Should we use `Record` or `Dictionary`?

**Review by delvedor — APPROVED** [review]
Overall LGTM, left a couple of nits :)

File: specification/specs/search/search/hits/Hit.ts (hunk 1 above)
Comment (Mpdreamz) [reply specification/specs/search/search/hits/Hit.ts:11]: For now yeah, these all indicate the .NET client does something special. Want to use these as markers to go over at some point

File: specification/specs/search/search/hits/Hit.ts (hunk 2 above)
Comment (Mpdreamz) [reply specification/specs/search/search/hits/Hit.ts:24]: both lol should we do `long | double | string` ?

Comment (delvedor) [reply specification/specs/search/search/hits/Hit.ts:24]: Yep! :)

File: output/typescript/types.ts (hunk 5 above)
Comment (Mpdreamz) [reply output/typescript/types.ts:146]: The input is `Dictionary` but the output is always `Record`

Comment (delvedor) [reply output/typescript/types.ts:146]: My bad, should have looked at the file name!
Some times this output is hidden, others not 😅

**Review by delvedor — APPROVED** [review]
LGTM

**Pull request merged**